docs/reference/protocol-spec.md

```diff
@@ -1,214 +1,251 @@
+
+The first byte in an RESP-serialized payload identifies its type.
+Following bytes constitute the type's contents.
+The `\r\n` (CRLF) is the protocol's **terminator**, which always separates its parts.
+
+The following table summarizes RESP's data types:
+
+| RESP data type | Minimal protocol version | Category | First byte |
+| --- | --- | --- | --- |
+| [Simple Strings](#simple-strings) | RESP2 | Simple | `+` |
+| [Errors](#errors) | RESP2 | Simple | `-` |
+| [Integers](#integers) | RESP2 | Simple | `:` |
+| [Bulk Strings](#bulk-strings) | RESP2 | Aggregate | `$` |
+| [Arrays](#arrays) | RESP2 | Aggregate | `*` |
+| [Nulls](#nulls) | RESP3 | Simple | `_` |
+| [Booleans](#booleans) | RESP3 | Simple | `#` |
+| [Doubles](#doubles) | RESP3 | Simple | `,` |
+| [Big Numbers](#big-numbers) | RESP3 | Simple | `(` |
+| [Blob Errors](#blob-errors) | RESP3 | Aggregate | `!` |
+| [Verbatim Strings](#verbatim-strings) | RESP3 | Aggregate | `=` |
+| [Maps](#maps) | RESP3 | Aggregate | `%` |
+| [Sets](#sets) | RESP3 | Aggregate | `~` |
+| [Pushes](#pushes) | RESP3 | Aggregate | `>` |
 
 <a name="simple-string-reply"></a>
 
-## RESP Simple Strings
-
-Simple Strings are encoded as follows: a plus character, followed by a string that cannot contain a CR or LF character (no newlines are allowed), and terminated by CRLF (that is "\r\n").
+### Simple Strings
+Simple Strings are encoded as follows: a plus (`+`) character, followed by a string.
+The string mustn't contain a CR or LF character (no newlines are allowed), and is terminated by CRLF (i.e., `\r\n`).
 
-Simple Strings are used to transmit non binary-safe strings with minimal overhead. For example, many Redis commands reply with just "OK" on success. The RESP Simple String is encoded with the following 5 bytes:
+Simple Strings transmit short, non-binary strings with minimal overhead.
+For example, many Redis commands reply with just "OK" on success.
+The encoding of this Simple String is the following 5 bytes:
 
-    "+OK\r\n"
+    +OK\r\n
 
-In order to send binary-safe strings, use RESP Bulk Strings instead.
+When Redis replies with a Simple String, a client library should return to the caller a string value composed of the first character after the `+` up to the end of the string, excluding the final CRLF bytes.
 
-When Redis replies with a Simple String, a client library should respond with a string composed of the first character after the '+'
-up to the end of the string, excluding the final CRLF bytes.
+For sending binary-safe strings, use [Bulk Strings](#bulk-strings) instead.
 
 <a name="error-reply"></a>
 
-## RESP Errors
-
-RESP has a specific data type for errors. They are similar to
-RESP Simple Strings, but the first character is a minus '-' character instead
-of a plus. The real difference between Simple Strings and Errors in RESP is that clients treat errors
-as exceptions, and the string that composes
-the Error type is the error message itself.
+### Errors
```

Comment:
So, following the same logic, we should have "Simple Error" and "Bulk Error" instead of "Error" and "Blob Error" to match strings. That's a nice consistency, but I wonder what would break if change names this far into the game.

Reply:
In [RESP3.md](https://github.com/redis/redis-specifications/blob/master/protocol/RESP3.md) it is "Simple Error" and "Blob Error".

I don't think it's a big deal to rename them or even mention both names (e.g. "bulk error AKA blob error"), but people will most certainly use both the old and the new terms forever, which is fine IMO as long as none of the terms are ambiguous.

Reply:
Fixed as "simple" and "bulk"